release_docs/NEWSLETTER.txt

```diff
@@ -1,3 +1,8 @@
+Release of HDF5 1.14.4 Library and Tools is now available from the HDF5 Releases page.

```

Comment:
This copies what is on the NEWSLETTER website.

Reply:
I figured the idea was to have a file that's empty other than the entries just so there's a quick reference to pull from. But, no complaints about the header and footer; just wanting to avoid putting version numbers anywhere they aren't exactly need since it just adds to the maintenance burden. If `h5vers` can handle it though, great!